workspaces/api-gateway/src/api/v1/webhooks/webhooks.controller.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Controller,
+  InternalServerErrorException,
+  Post,
+  RawBodyRequest,
+  Req,
+} from '@nestjs/common';
+import Stripe from 'stripe';
+import { InjectRepository } from '@nestjs/typeorm';
+import type { Repository } from 'typeorm';
+import type { Request } from 'express';
+
+import { Payment } from 'database/models/payment.entity';
+import { stripe } from 'database/models/paymentMethods/stripeMethod.entity';
+import type { IMetadataParams } from 'api/v1/payments/payments.interface';
+
+@Controller('webhook')
+export class WebhooksController {
+  private stripe: Stripe;
+  private endpointSecret: string;
+
+  constructor(
+    @InjectRepository(Payment)
+    private paymentRepository: Repository<Payment>,
+  ) {
+    this.stripe = stripe;
+    this.endpointSecret = process.env.SECRET_ENDPOINT_STRIPE;
+  }
+
+  @Post('/stripe')
+  public async createCheckoutStripe(
+    @Req() request: RawBodyRequest<Request>,
+  ): Promise<Stripe.Event> {
+    const payload = request.rawBody;
+    const sig = request.headers['stripe-signature'];
+    let event: Stripe.Event;
+
+    try {
+      event = this.stripe.webhooks.constructEvent(
+        payload,
+        sig,
+        this.endpointSecret,
+      );
+    } catch (err) {
+      throw new InternalServerErrorException(err);
+    }
+
+    switch (event.type) {
+      case 'charge.succeeded': {
+        const session = event.data.object;
```

Comment:
should move this one outside the switch case because all of cases use this one.

Convert to the camelCase